### DISCUSSION TITLE
Change feed processor discriminator

### DISCUSSION THREAD

**Discussion by OpsOwns** [Azure/azure-cosmos-dotnet-v3#2743] [Q&A] [answered]
It's possible to block some how receiving unwanted objects from container. For example if I got class StoredDocument  and this has got field  paylod this is json string and apart from it has property discriminator with custom type of class  and i want to in my change feed processor get only indicated types by this property. There is possible block somehow unwanted types  because right now i have got something like that.

```csharp
   public async Task HandleDocuments(IReadOnlyCollection<StoredDocument > documents, CancellationToken token)
        {
            foreach (var document in documents.Where(x => !x.Discriminator.Contains("product", StringComparison.InvariantCulture)))
            {
}
}
```

**Comment by bartelink** [comment, marked as answer]
The fundamental constraint is that there is no server-side filtering capability in the endpoint that the CFP uses to obtain its data - it's a "fetch all docs since this continuation token" style query versus a filtered one, and it would be a pretty fundamental change to be able to also supply a filter. Because of the huge variety of filtering and processing that different use cases demand, and the need to allow users to control the parsing (or not) of documents, doing a client-side `Where` is about as appropriate an API as possible IMO.

[Tiny bit related feature idea](https://github.com/Azure/azure-cosmos-dotnet-v3/issues/1763)

**Reply by OpsOwns** [reply]
So i have to filter out unwanted documents after received documents ?

**Reply by bartelink** [reply]
Yes, the current API doesn't offer a way for you to specify for filtering to happen, so filtering the supplied list of items is the way to go.
